test/dependency-manager/interpolation-validation.test.ts

```diff
@@ -0,0 +1,145 @@
+import { expect } from 'chai';
+import { buildSpecFromYml, ValidationErrors } from '../../src';
+import { registerInterpolation } from '../../src/dependency-manager/utils/interpolation';
+
+describe('interpolation-validation', () => {
+  const context = {
+    architect: {
+      tag: 'latest'
+    }
+  }
+
+  describe('validate build block', () => {
+    it('cannot use secret in build block', async () => {
+      const component_config = `
+        name: hello-world
+        secrets:
+          environment: prod
+        services:
+          api:
+            build:
+              args:
+                ENV: \${{ secrets.environment }}
+        `
+
+      const component_spec = buildSpecFromYml(component_config)
+
+      expect(() => {
+        registerInterpolation(component_spec, context)
+      }).to.be.throws(ValidationErrors);
+    });
+
+    it('cannot use conditional in build block', async () => {
+      const component_config = `
+        name: hello-world
+        services:
+          api:
+            build:
+              args:
+                \${{ if architect.environment == 'local' }}:
+                  ENV: prod
+        `
+
+      const component_spec = buildSpecFromYml(component_config)
+
+      expect(() => {
+        registerInterpolation(component_spec, context)
+      }).to.be.throws(ValidationErrors);
+    });
+
+    it('cannot use conditional around build block', async () => {
```

Comment:
This seems like it would be problematic in a lot of cases as it was a replacement for the debug block. We use conditionals around `build` blocks for some of our own services as it is so I feel like it's really likely that this is something other users would need as well

Reply:
The thought is that you could still use conditionals anywhere, but not around the build block if they are deploy-time specific.

So the following wouldn't be allowed:
`${{ if architect.environment == 'local' }}:`

However, this would:
`${{ if architect.build.tag == 'local' }}:`
See: https://github.com/architect-team/architect-cli/pull/580/files#diff-ccf91dc9ee7a6fcf782de9f37462393ebf1f28f1a01e586ecb6e3bbd97e1034bR93

----

Still trying to decide if we make a special case for `architect.environment == 'local'` for some period of time.